modules/ROOT/pages/fd-iterating.adoc

```diff
@@ -0,0 +1,112 @@
+
+Suppose that, as in the previous example, you use a Set Variable component just before the scope of a Parallel For-Each component. Within the Set Variable component, you set the variable `NumOfElements` to 0. You then use a Set Variable component in the Parallel For-Each scope, using the same variable, that specifies a DataWeave script that increments the value by 1 every time it is called. For each parallel process that the component runs, the highest value reached by `NumOfElements` is 1. And if subsequent cards in the flow use `NumOfElements`, its initial value after the Parallel For-Each component finishes processing the collection is 0, which is the same value it had before the Parallel For-Each component started.
+
+
+== Error Handling
+
+Both the For-Each component and Parallel For-Each component stop processing a collection if an element causes any component, connector, or module in a scope to throw an exception. In addition, the Parallel For-Each component stops processing a collection if one of the parallel processes times out while waiting for an open connection to a system. You can set the maximum length of time that a parallel process can wait.
+
+When the processing of a collection stops because of an exception, Flow Designer displays a red circle on the card that threw the exception. When the processing of a collection stops because of a timeout, Flow Designer displays a red circle on the Parallel For-Each component's card. You can open the *Logs* pane to read the error message.
+
+== Configuring a For-Each Component
+
+When you configure this component, you must specify values in these fields:
+
+Collection:: An expression that returns a Java collection, object array, map, or DOM nodes. The default is `payload`.
+
+Batch Size:: Partitions the collection into sub-collections of the specified number of elements. For example, if a collection has 200 elements and you set the batch size to 50 elements, the component processes 4 batches of 50 elements each.
+
+Root Message Variable Name:: Name of the property that stores the entire message -- payload, variables, and attributes -- that was passed to a For-Each component from the previous card in a flow. 
+
+Counter Variable Name:: Name of the property that stores the index of the element that is being processed. For example, if a collection is `['a','b','c']`, the index of `a` is 0, of `b` 1, and of `c` 2. When the For-Each component is processing the second element, the value of the variable is 1.
+
+
+== Configuring a Parallel For-Each Component
+
+When you configure this component, you must specify values in these fields:
+
+Collection:: Specifies the expression that defines the collection of parts to be processed in parallel. The default is `payload`.
+
+Timeout:: Specifies the maximum time in milliseconds that a parallel process can wait for an open connection to a system reached via a connector.
```

Comment:
This is the timeout of the execution of the whole scope in the parallel foreach. It can timeout at any time, in any card. That is, the timeout there is the maximum time the for each can take to execute the scope.